### ISSUE TITLE
Handle running out of quota / rate limit

### ISSUE THREAD

**Issue by jeffmendoza** [ossf/allstar#170]
Allstar aggressively caches and saves the cache between runs, however it is still running out of quota on orgs with a large number of small repositories. For some reason the cache is not helping avoid this from run to run. We'll need to figure out what is going on with the cache, and probably implement a proper rate limit delay to pick up where we left off.

**Comment by yorinasub17** [comment]
>  For some reason the cache is not helping avoid this from run to run.

I looked into this and traced it down to because of the ordering of the transports. The `httpcache` transport uses the request headers to determine whether or not to return the cached responses, and because the `httpcache` transport is run first, this is ALWAYS different.

Specifically, what happens is that the `httpcache` transport records the original request headers as new headers prefixed with `X-Varied-HEADER_KEY` headers on the response. However, this happens after the GH `RoundTrip` function is called, which means that when `httpcache` records the request headers, it includes the `Authorization` token from the underlying GH transport. Well this doesn't match the next time the cache lookup happens because the request checking happens BEFORE calling `RoundTrip` on the underlying transport, which means the request header is missing the `Authorization` token.

Once I reordered the transports, I was able to observe the cache being utilized much more. Here is the PR with this! https://github.com/ossf/allstar/pull/189
